run.cc

```diff
@@ -60,20 +62,179 @@ void runTests() {
+  
+  cout<< "\n";
+  
+  cout<< "Number of bins that deviate 2*sigma for Exp A and Exp D: " << datA.checkCompatibility(datD,2) << endl;
+  
+  cout<< "\n";
+  
+  cout<< " The number of datapoints that deviate from 2*sigma is 0 and the maximum number of datapoints that are allowed to deviate from 2*sigma are 3( ~5% of total datapoints) "<< endl;
+  
+  cout<< "\n";
+  
+  
+  cout<< "Number of bins that deviate 2*sigma for Exp B and Exp C: " << datB.checkCompatibility(datC,2) << endl;
+  
+  cout<< "\n";
+  
+  cout<< " The number of datapoints that deviate from 2*sigma is 1 and the maximum number of datapoints that are allowed to deviate from 2*sigma are 3( ~5% of total datapoints) "<< endl;
+  
+  cout<< "\n";
+  
+  
+  cout<< "Number of bins that deviate 2*sigma for Exp B and Exp D: " << datB.checkCompatibility(datD,2) << endl;
+  
+  cout<< "\n";
+  
+  cout<< " The number of datapoints that deviate from 2*sigma is 4 and the maximum number of datapoints that are allowed to deviate from 2*sigma are 3( ~5% of total datapoints). "<< endl;
+  
+  cout<< "\n";
+  
+  cout<< "Number of bins that deviate 2*sigma for Exp C and Exp D: " << datC.checkCompatibility(datD,2) << endl;
+  
+  cout<< "\n";
+  
+  cout<< " The number of datapoints that deviate from 2*sigma is 1 and the maximum number of datapoints that are allowed to deviate from 2*sigma are 3( ~5% of total datapoints). "<< endl;
+  
+  cout<< "\n";
+  //Exercise 2(first subpart):
+  /*cout<< "The background is :"  << endl;
+  std::vector<double> fu=datA.fun_f();
+  for (int i=0; i< fu.size(); i++)
+  {
+    cout<< fu[i] << "\n";
+  }*/
+  
+ // Exercise 2b
+  cout<<"The X^2/ndf of Exp A :" << endl;
+  std::vector<double> ch=datA.func_X();
+  for (int i=0; i< ch.size(); i++)
+  {
+    cout<< ch[i]/52 << "\n";
+  }
+  cout<< "The X^2 distribution for Exp A is a good fit." <<endl; 
+  
+  cout<< "\n";
+  
+  cout<<"The X^2/ndf of Exp B :" << endl;
+  std::vector<double> ch2=datB.func_X();
+  for (int i=0; i< ch2.size(); i++)
+  {
+    cout<< ch2[i]/52 << "\n";
+  }
+  cout<< "The X^2 distribution for Exp B is best fit." <<endl;
+  
+  cout<< "\n";
+  
+  cout<<"The X^2/ndf of Exp C :" << endl;
+  std::vector<double> ch3=datC.func_X();
+  for (int i=0; i< ch3.size(); i++)
+  {
+    cout<< ch3[i]/52 << "\n";
+  }
+  cout<< "The X^2 distribution for Exp B is a best fit." <<endl;
+  
+  cout<< "\n";
+  
+  cout<<"The X^2/ndf of Exp D :" << endl;
+  std::vector<double> ch4=datD.func_X();
+  for (int i=0; i< ch4.size(); i++)
+  {
+    cout<< ch4[i]/52 << "\n";
+  }
+  cout<< "The X^2 distribution for Exp B is a best fit." <<endl;
+  
+  cout<< "\n";
+  //Exercise-2c
+  
+  Data avgAB = datA.average(datB);
+  Data avgCD = datC.average(datD);
+  
+  Data avgT = avgAB.average(avgCD);
+  
+  cout<< "The X^2/ndf of combined data set is: " << avgT.func_X()[0]/52.0<<endl;
+  
+  cout<< "The X^2 distribution for combined data set suggests wrong hypothesis." <<endl;
+  
```

Comment:
Correct, but do you know why? What criterion did you use to arrive at this statement?